core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -280,7 +283,8 @@ bool thread_enable_prealloc_rpc_cache(void)
 	}
 
 	rv = true;
-	thread_prealloc_rpc_cache = true;
+	if (IS_ENABLED(CFG_PREALLOC_RPC_CACHE))
```

Comment:
In case this is true we can return true directly at the beginning of the function without any other checks or locks.

Reply:
you're rigth.